pkg/querier/querier.go

```diff
@@ -93,6 +98,9 @@ func (q *Querier) SelectLogs(ctx context.Context, params logql.SelectLogParams)
 	var err error
 	params.Start, params.End, err = q.validateQueryRequest(ctx, params)
 	if err != nil {
+		if errors.Is(err, errQueryRangeOutsideLookbackPeriod) {
+			return iter.NoopIterator, err
```

Comment:
I'm not clear why we need to return NoopIterators. If the error is present, which it is, we'll just return that, no? If we've otherwise adjusted the bounds to be within the valid range, then this won't error anyways.

Reply:
I think it's because it's not erroring but returning empty result, Honestly I'm not sure, I feel like an error might be better for the user so that he understand why he has no result (400).

I understand that when it's overlap, it's fine to return partial result.

Interesting that we had this bug only in the querier and not the frontend but reading it seems legit.

Reply:
The one instance when I saw this happen was when a user did a query that had a start time older than allowed lookback and the step was `23h`. QF correctly changed the start time to be within the lookback period and then the step align code moved the whole interval from a split to before the lookback period.

I think in this case we should handle it gracefully instead of erroring the user, same as how we do the adjustment of query interval when it is out of lookback period.